src/Config/ObjectModel/RuntimeEntities.cs

```diff
@@ -121,9 +121,28 @@ private static Entity ProcessRestDefaults(Entity nameCorrectedEntity)
         // If no Rest node was provided in the config, set it with the default state of enabled for all verbs
         if (nameCorrectedEntity.Rest is null)
         {
-            nameCorrectedEntity = nameCorrectedEntity
-                with
-            { Rest = new EntityRestOptions(EntityRestOptions.DEFAULT_SUPPORTED_VERBS) };
+            nameCorrectedEntity = (nameCorrectedEntity.Source.Type is EntitySourceType.StoredProcedure) ?
+                                    nameCorrectedEntity
+                        with
+                                    { Rest = new EntityRestOptions(EntityRestOptions.DEFAULT_SUPPORTED_VERBS_FOR_SP) }
+                    :
+                    nameCorrectedEntity
+                        with
+                    { Rest = new EntityRestOptions(EntityRestOptions.DEFAULT_SUPPORTED_VERBS) };
+        }
+        else
+        {
+            if (nameCorrectedEntity.Rest.Methods.Length == 0)
```

Comment:
Add a comment that calls out that we only care about this extra check when length == 0 because the `EntityRestOptionsConverter` properly parses any user specified methods.